src/app-gocardless/banks/integration-bank.js

```diff
@@ -37,7 +37,14 @@ export default {
   },
 
   normalizeTransaction(transaction, _booked) {
-    return transaction;
+    return {
+      ...transaction,
+      date:
+        transaction.bookingDate ||
+        transaction.bookingDateTime ||
+        transaction.valueDate ||
+        transaction.valueDateTime,
+    };
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think the ordering used here will cause a regression. UI code: https://github.com/actualbudget/actual/pull/1499/files

```suggestion
      date:
        transaction.valueDate ||
        transaction.valueDateTime ||
        transaction.bookingDate ||
        transaction.bookingDateTime,
    };
```

Reply:
It's a change in behavior, but I don't see how it would be a regression since we're switching to the expected order as documented by GoCardless and which you seemed okay with here: https://github.com/actualbudget/actual-server/pull/237#issuecomment-1670822085

I'd also be fine with using the minimum as you suggested, but I figured that using the order that I'd expect according to their documentation would be more semantically correct.

Reply:
Apologies; poor choice of words of my behalf. Yes, this isn't a regression, but a change in functionality.

Conceptually I agree with the order you are proposed, but I'm worried about changing the functionality unless there is a good reason for the change as it introduces a certain factor of risk? If there is no specific reason to change the logic (bug/issue) - I'd err on the side of keeping the order as-is until we find a good reason for the change. Thus having it more stable.

WDYT?

Reply:
The one risk is that some banks that violate their documentation will have transactions returned with a later date than previously (by a day or two), but on the other hand banks that _don't_ violate their documentation will start returning the correct data which they until now haven't.

So while I don't necessarily have a problem with leaving it as-is it means that we're choosing to give priority to users of banks where the API returns 'incorrect' data over users of banks where the API returns 'correct' data.

Which to me seems strange when the only thing that the affected users will experience is that their imported transactions, which they haven't manually previously created a transaction that the imported one will be merged into, will be off by a day or two.  I would argue that this would also be a good hint to them that they should file a request for an integration with their bank to be added so that its incorrect behavior can be explicitly normalized rather than just accidentally working at the expense of banks that do the 'correct' thing.

---

Ultimately I'm planning on adding integrations for all banks that I care about so it won't affect me directly, and so I don't really care too much about which way we go other than from a theoretical correctness perspective.

Reply:
(I say that, but as written this is actually wrong I just noticed.  The `DateTime` variants need to be parsed since we expect only dates.)

Reply:
Hmm.. ok, you convinced me. Worst case scenario: this breaks in which case we can quickly patch it.

But can we then make all the banks follow this order? For example: ing-pl-ingbplpw is following the old ordering. Whichever way we choose - we should make it consistent.

Reply:
Since the previous integrations (ing-pl-ingbplpw and mbank-retail-brexplpw) were added as part of the initial Nordigen support I assumed that the old order was the correct one for those banks, and that they'd have either added transaction normalization or just defaulted to the other order if it wasn't the correct order for them.  I don't have any example transactions from those banks to go off of though so I can't be sure, and so I can change it to the new order if you'd like.

Reply:
I would say if we're changing this logic in one place, then lets change it in all. Lets keep the code consistent. If after merging someone reports this as being the wrong call - we can always swap back. But I doubt it will be this case.

Reply:
Done.